kvdb/postgres/db.go

```diff
@@ -0,0 +1,233 @@
+// +build kvdb_postgres
+
+package postgres
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"io"
+	"sync"
+	"time"
+
+	"github.com/btcsuite/btcwallet/walletdb"
+	_ "github.com/jackc/pgx/v4/stdlib"
+)
+
+const (
+	// hexTableId is the table name for top-level buckets data under a key
+	// that cannot be mapped to a distinct table.
+	hexTableId = "hex"
+
+	// sequenceTableId is the table name that holds top-level bucket
+	// sequence numbers.
+	sequenceTableId = "sequence"
+
+	// systemTablePrefixExtension is the prefix given to system tables.
+	systemTablePrefixExtension = "sys"
+)
+
+// KV stores a key/value pair.
+type KV struct {
+	key string
+	val string
+}
+
+// db holds a reference to the postgres connection connection.
+type db struct {
+	// cfg is the postgres connection config.
+	cfg *Config
+
+	// prefix is the table name prefix that is used to simulate namespaces.
```

Comment:
Shouldn't this be a top-level argument in the main config? As it would allow multiple `lnd` nodes to share the same postgres database.

Reply:
What is the gain of having multiple lnds write to the same db in diff namespaces? I thought it is safer to keep them isolated. For a cluster of nodes that operate as one, namespace isolation is probably not necessary?

Reply:
> What is the gain of having multiple lnds write to the same db in diff namespaces? I thought it is safer to keep them isolated.

It would make it possible to run _multiple_ `lnd` nodes with a single database instance vs having to maintain/backup/monitor multiple DB server isntances.

Reply:
Oh, but there is also the `database` level in a database instance. The `database` is something that can be configured in `lnd.conf`, so it is already possible to point multiple `lnd` nodes to the same postgres instance. The itests also run like that. Each test creates its own random database in the same postgres instance.

Reply:
Makes sense, had initially missed that!